source/server/options_impl.cc

```diff
@@ -270,7 +270,14 @@ OptionsImpl::OptionsImpl(std::vector<std::string> args,
   drain_time_ = std::chrono::seconds(drain_time_s.getValue());
   parent_shutdown_time_ = std::chrono::seconds(parent_shutdown_time_s.getValue());
   socket_path_ = socket_path.getValue();
-  StringUtil::atoull(socket_mode.getValue().c_str(), reinterpret_cast<uint64_t&>(socket_mode_), 8);
+
+  if (socket_path_.at(0) == '@') {
+    socket_mode_ = 0;
```

Comment:
Does something fail if you leave it as `600` for an abstract namespace socket? If not, I think you can just omit this. Another idea is to construct the `PipeInstance` here, which includes a mode, so that you only have to pass one argument through the several function calls, instead of both address and mode.

Reply:
Yes, It's failing. `PipeInstance` need 0 as mode if it's in abstract namespace otherwise throw an exception like https://github.com/envoyproxy/envoy/blob/02e97e1dcd948c14b62de95b348badb5ee6d1575/source/common/network/address_impl.cc#L280

Reply:
Got it. It's fine to leave as-is then.